drivers/azure/nic.go

```diff
@@ -0,0 +1,132 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+func (p *provider) getSubnetsClient() (network.SubnetsClient, error) {
+	subnetsClient := network.NewSubnetsClient(p.subscriptionId)
+	auth, err := auth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		return subnetsClient, err
+	}
+	subnetsClient.Authorizer = auth
+	return subnetsClient, nil
+}
+
+// getVirtualNetworkSubnet returns an existing subnet from a virtual network
+func (p *provider) getVirtualNetworkSubnet(ctx context.Context, vnetName string, subnetName string) (network.Subnet, error) {
+	subnetsClient, err := p.getSubnetsClient()
+	if err != nil {
+		return network.Subnet{}, err
+	}
+	return subnetsClient.Get(ctx, p.resourceGroup, vnetName, subnetName, "")
+}
+
+
+func (p *provider) getIPClient() (network.PublicIPAddressesClient, error) {
+	ipClient := network.NewPublicIPAddressesClient(p.subscriptionId)
+	auth, err := auth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		return ipClient, err
+	}
+	ipClient.Authorizer = auth
+	return ipClient, nil
+}
+
+// CreatePublicIP creates a new public IP
+func (p *provider) createPublicIP(ctx context.Context, ipName string) (ip network.PublicIPAddress, err error) {
```

Comment:
is creating a static public IP necessary?  can we just use the ephemeral public IP address associated with the instance? This is what we do for other cloud providers.